File: ios/C_to_RN/C_to_RN.m
#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
@interface RCT_EXTERN_MODULE(C_to_RN, NSObject)
RCT_EXTERN_METHOD(WrapFactorial: (int)num callback: (RCTResponseSenderBlock)callback;)
@end

File: ios/C_to_RN/C_to_RN.swift
import Foundation

@objc(C_to_RN)
class C_to_RN : NSObject {
  
  @objc static func requiresMainQueueSetup() -> Bool {return false }
  @objc func WrapFactorial(_ num:(Int), callback: RCTResponseSenderBlock) {
       callback([NSNull(), factorial(Int32(num)) as Any])
  }
  
}
